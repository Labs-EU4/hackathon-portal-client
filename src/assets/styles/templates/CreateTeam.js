import styled from "styled-components";

import BodyContainer from "../atoms/BodyContainer";

export const BodyRow = styled(BodyContainer)`
  flex-direction: column;
  align-items: flex-start;
  height: 100%;
  width: 100%;
  max-width: 100vw;
`;

export const BodyColumn = styled(BodyContainer)`
  flex-direction: row;
  align-items: flex-start;
  height: 100%;
  width: 100%;
  max-width: 100vw;
`;

export const Form = styled.form`
  background-color: white;
  width: 50%;
  height: 35vh;
  border: 1px solid ${props => props.theme.color.grey.border};
  display: flex;
  flex-direction: column;
  align-items: center;
  margin-left: 25%;
  border-radius: 6px;
  h4 {
    margin: 10px;
    padding: 5px;
  }
  label {
    margin: 10px;
  }
  input {
    width: 50%;
    padding: 10px;
    margin: 10px;
    border: 1px solid ${props => props.theme.color.grey.border};
    border-radius: 6px;
  }
  button {
    width: 50%;
    margin: 10px;
  }
`;