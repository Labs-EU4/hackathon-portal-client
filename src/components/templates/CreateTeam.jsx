import React, { useEffect } from "react";
import { useParams, useHistory } from "react-router-dom";
import { useSelector, useDispatch } from "react-redux";
import { createTeamName } from "../../store/participantTeams/actions";
import { Formik } from "formik";
import {
  BodyRow,
  BodyColumn,
  Form,
  Title
} from "../styles/templates/CreateTeamStyling";
import { Footer } from "../organisms/index";
import UserHeader from "../organisms/UserHeader";
import { RowHead } from "../atoms/RowHead";
import { H3 } from "../atoms/Heading";
import Button from "../atoms/Button";
import TeamView from "./TeamView";
import { useTeams } from "../../hooks";
import WideBody from "../atoms/WideBody";
import Nav from "../molecules/Nav";

const CreateTeam = () => {
  const dispatch = useDispatch();
  const history = useHistory();
  const { id } = useParams();
  const { userId } = useSelector(state => state.currentUser);
  const [teams, fetchTeams] = useTeams(id);
  const team = teams.find(t => t.team_lead === userId);
debugger
  useEffect(() => {
    fetchTeams();
  }, [fetchTeams]);

  const handleTeamSubmit = values => {
    const teamData = {
      team_name: values.team_name,
      eventId: id
    };
    dispatch(createTeamName(teamData, history));
  };
debugger
  const { event_title } = useSelector(state =>
    state.events.data.find(event => event.id === Number(id))
  );
debugger
  return (
    <div>
      <UserHeader />
      <WideBody>
        <Nav />
        <BodyRow>
          <RowHead>
            <H3>Participant Teams</H3>
          </RowHead>
          <BodyColumn>
            {!team ? (
              <Formik
                initialValues={{ team_name: "" }}
                onSubmit={handleTeamSubmit}
              >
                {props => (
                  <Form onSubmit={props.handleSubmit}>
                    <h4>
                      You are creating a team for <Title>{event_title}</Title>
                    </h4>
                    <label>Team Name</label>
                    <input
                      type="text"
                      onChange={props.handleChange}
                      onBlur={props.handleBlur}
                      value={props.values.name}
                      name="team_name"
                    />
                    <Button type="submit" color="green">
                      Submit
                    </Button>
                  </Form>
                )}
              </Formik>
            ) : (
              <TeamView {...{ team }} />
            )}
          </BodyColumn>
        </BodyRow>
      </WideBody>
      <Footer />
    </div>
  );
};

export default CreateTeam;
