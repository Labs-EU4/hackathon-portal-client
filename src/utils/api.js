import Axios from "axios";
import { toast } from "react-toastify";
import { resetUser } from "../store/user/actions";

const baseURL = process.env.REACT_APP_API_URL;

// if process.env.CONTEXT = "production" {
//   Point to heroku API
// }
// if process.env.BRANCH = "develop" {
  // point to Heroku staging API
// }
// if process.env.CONTEXT = "deploy-preview" {
//   point to deploy-preview HEROKU API (the separate one)
// }



export const axiosWithAuth = token =>
  Axios.create({
    baseURL,
    headers: {
      Authorization: token
    }
  });

export const axios = Axios.create({ baseURL });

export const selectToken = state => state.currentUser.token;

export const showError = message => {
  toast.error(message);
};

export const showSuccess = message => {
  toast.success(message);
};

export function* handleError({ response }, put, history) {
  const { message, statusCode } = response.data;
  if (statusCode === 404) {
    history.push("/not-found");
  }
  if (message === "jwt expired") {
    yield put(resetUser());
  } else showError(`⚠️ ${message}`);
}
