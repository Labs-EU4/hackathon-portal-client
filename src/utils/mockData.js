export const initialState = {
  events: {
    data: [
      {
        id: 1,
        event_title: "EuroHack",
        event_description: "the best event hack hack hack macbook",
        creator_id: 1,
        start_date: "10/09/11",
        end_date: "08/07/13",
        location: "Europe",
        guidelines: "cool hekhkdsjfhsdkjf skdjfhskdjfhkjsdfh skjdfhksjdfh",
        participation_type: "Team",
        category_id: 1,
        requirements: [
          {
            video_url:
              "https://res.cloudinary.com/ibkrice-com/video/upload/v1581259780/Harvesting_ifato8.mp4"
          }
        ],
        ProfileImage: [
          {
            image_url:
              "https://res.cloudinary.com/ibkrice-com/video/upload/v1581259780/Harvesting_ifato8.mp4"
          }
        ],
        rubrics: [
          "Presentation",
          "Product Fit",
          "Functionality",
          "Innovation",
          "Product Design",
          "Extensibility"
        ]
      },
      {
        id: 2,
        event_title: "World",
        event_description: "number one hacker event in the world",
        creator_id: 1,
        start_date: "07/09/11",
        end_date: "08/07/13",
        location: "Asia",
        guidelines: "come well prepared,no cheating,no fighting",
        participation_type: "Team",
        category_id: 2
      }
    ],
    categories: [
      { id: 2, category_name: "Summer Hackton" },
      { id: 3, category_name: "World Hackton" },
      { id: 3, category_name: "Asia Hackton" },
      { id: 3, category_name: "Gamer's Hackton" },
      { id: 3, category_name: "Advanced Hackton" }
    ]
  },

  currentUser: {
    token: 347534574536,
    userId: 1,
    fullname: "Jake Tom",
    email: "8omemail@google.com",
    username: "Jake22",
    bio: "I like coding ",
    image: ""
  },
  submissions: [
    {
      id: 1,
      project_title: "Mock Project",
      participant_or_team_name: "Mock Team",
      event_id: 7,
      video_url: "",
      git_url: "https://www.google.com/",
      project_writeups: "No mock write up",
      submitted_by: 14,
      created_at: "2020-03-02",
      updated_at: "2020-03-02",
      average_rating: 0,
      acted_judges: 0,
      number_of_judges: 0
    }
  ],

  teams: [
    { id: 1, team_name: "The A team", team_lead: 2 },
    { id: 2, team_name: "Hacking central", team_lead: 3 },
    { id: 3, team_name: "blackhats", team_lead: 4 },
    { id: 4, team_name: "Spies", team_lead: 1 }
  ],
  participant_team_members: [
    {
      team_member: 3,
      team_id: 1
    },
    {
      team_member: 4,
      team_id: 1
    },
    {
      team_member: 5,
      team_id: 1
    }
  ]
};
