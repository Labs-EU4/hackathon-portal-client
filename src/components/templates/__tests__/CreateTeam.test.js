import React from "react";
import { Router } from "react-router";
import { createMemoryHistory } from "history";
import { Provider } from "react-redux";
import { render } from "@testing-library/react";
import configureStore from "redux-mock-store";
import "@testing-library/jest-dom/extend-expect";
import CreateTeam from "../CreateTeam";
import { initialState } from "../../../utils/mockData";

const history = createMemoryHistory();

let component;
let mockStore;
let store;

jest.mock("react-router-dom", () => ({
  ...jest.requireActual("react-router-dom"), // use actual for all non-hook parts
  useParams: () => ({
    id: 1
  }),
  useRouteMatch: () => ({ url: "/dashboard/event/:id/participant_submission" })
}));

beforeEach(() => {
  mockStore = configureStore();
  store = mockStore(initialState);
  component = render(
    <Router history={history}>
      <Provider store={store}>
        <CreateTeam />
      </Provider>
    </Router>
  );
});

describe("Shows all the text nodes on CreateTeam.js that are contained on the making an event form", () => {
  it("CreateTeam.js component renders properly", () => {
    expect(component).toMatchSnapshot();
  });
  it("The text node for the first character of the current user mail, renders properly on the menu", () => {
    expect(component.queryByText(/8/i)).toBeInTheDocument();
  });
  it("The text node for the <h3> Participant Teams text node, renders properly", () => {
    expect(component.queryByText(/Participant Teams/i)).toBeInTheDocument();
  });
  it("The text node for the <h4> You are creating a team for text node, renders properly", () => {
    expect(
      component.queryByText(/You are creating a team for/i)
    ).toBeInTheDocument();
  });
  it("The text node for the team being created for the selected event 'EuroHack' <span>, renders properly", () => {
    expect(component.queryByText(/EuroHack/i)).toBeInTheDocument();
  });
  it("The text node for the Team Name <label> input, renders properly", () => {
    expect(component.queryByText(/Team Name/i)).toBeInTheDocument();
  });
  it("The role for the button used to submit the form to create a team renders properly", () => {
    expect(component.getByRole(/button/i)).toBeInTheDocument();
  });
  // it("The text node for the submit button <button>, renders properly", () => {
  //   expect(component.getByText(/submit/i)).toBeInTheDocument();
  // });
});
