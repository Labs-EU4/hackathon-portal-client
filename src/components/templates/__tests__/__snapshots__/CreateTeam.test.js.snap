// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Shows all the text nodes on CreateTeam.js that are contained on the making an event form CreateTeam.js component renders properly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <div
          class="sc-caSCKo bhjPEW"
        >
          <a
            class="sc-TOsTZ kTZUxM"
            href="/"
          >
            <img
              alt="Hackton - Organise hackathons"
              class="sc-fBuWsC kyHoeo"
              src="Hackton-logo.png"
            />
          </a>
          <div
            class="sc-cHGsZl fQyjuM"
          >
            <div
              class="sc-gisBJw dYkUDe"
            >
              <button
                class="sc-gPEVay jqeTh sc-kjoXOD dmrfxM"
                color="grey"
                data-testid="label"
              >
                Log Out
              </button>
            </div>
          </div>
        </div>
        <div
          class="sc-htpNat hiQpHM"
        >
          <div
            class="sc-bdVaJa sc-bxivhb kGXkcm"
          >
            <div
              class="sc-kgAjT eagLfN"
            >
              <h3
                class="sc-kgoBCf gxiaSr"
              >
                Participant Teams
              </h3>
            </div>
            <div
              class="sc-bdVaJa sc-ifAKCX eTxUZm"
            >
              <form
                class="sc-EHOje gcwXic"
              >
                <h4>
                  You are creating a team for 
                  <span
                    class="sc-bZQynM iIlkKe"
                  >
                    EuroHack
                  </span>
                </h4>
                <label>
                  Team Name
                </label>
                <input
                  name="team_name"
                  type="text"
                  value=""
                />
                <button
                  class="sc-gPEVay cgKgRb"
                  color="green"
                  type="submit"
                >
                  Submit
                </button>
              </form>
            </div>
          </div>
        </div>
        <div
          class="sc-gZMcBi jKbyTj"
        >
          <p
            class="sc-fjdhpX bjMPwO"
          >
            International Crafters © 2020
          </p>
          <div>
            <a
              class="sc-gqjmRU sc-jTzLTM kFrNrp"
              href="#"
            >
              Privacy
            </a>
            <a
              class="sc-gqjmRU sc-jTzLTM kFrNrp"
              href="#"
            >
              Contacts
            </a>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <div
        class="sc-caSCKo bhjPEW"
      >
        <a
          class="sc-TOsTZ kTZUxM"
          href="/"
        >
          <img
            alt="Hackton - Organise hackathons"
            class="sc-fBuWsC kyHoeo"
            src="Hackton-logo.png"
          />
        </a>
        <div
          class="sc-cHGsZl fQyjuM"
        >
          <div
            class="sc-gisBJw dYkUDe"
          >
            <button
              class="sc-gPEVay jqeTh sc-kjoXOD dmrfxM"
              color="grey"
              data-testid="label"
            >
              Log Out
            </button>
          </div>
        </div>
      </div>
      <div
        class="sc-htpNat hiQpHM"
      >
        <div
          class="sc-bdVaJa sc-bxivhb kGXkcm"
        >
          <div
            class="sc-kgAjT eagLfN"
          >
            <h3
              class="sc-kgoBCf gxiaSr"
            >
              Participant Teams
            </h3>
          </div>
          <div
            class="sc-bdVaJa sc-ifAKCX eTxUZm"
          >
            <form
              class="sc-EHOje gcwXic"
            >
              <h4>
                You are creating a team for 
                <span
                  class="sc-bZQynM iIlkKe"
                >
                  EuroHack
                </span>
              </h4>
              <label>
                Team Name
              </label>
              <input
                name="team_name"
                type="text"
                value=""
              />
              <button
                class="sc-gPEVay cgKgRb"
                color="green"
                type="submit"
              >
                Submit
              </button>
            </form>
          </div>
        </div>
      </div>
      <div
        class="sc-gZMcBi jKbyTj"
      >
        <p
          class="sc-fjdhpX bjMPwO"
        >
          International Crafters © 2020
        </p>
        <div>
          <a
            class="sc-gqjmRU sc-jTzLTM kFrNrp"
            href="#"
          >
            Privacy
          </a>
          <a
            class="sc-gqjmRU sc-jTzLTM kFrNrp"
            href="#"
          >
            Contacts
          </a>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
